src/dtsh/io.py

```diff
@@ -309,3 +309,28 @@ def readline(self, multi_prompt: Optional[Sequence[Any]] = None) -> str:
             return line.rstrip()
         self._in.close()
         raise EOFError()
+
+
+class DTShInputCmds(DTShInput):
+    """Command-line input processor.
+
+    Uses a list of shell commands provided via command line options.
+    """
+
+    _cmds: List[str]
+
+    def __init__(self, cmds: List[str]) -> None:
+        """Initialize object.
+
+        Args:
+            cmds: List of commands to execute.
+        """
+        self._cmds = cmds
+
+    def readline(self, multi_prompt: Optional[Sequence[Any]] = None) -> str:
+        """Overrides DTShInput.readline()."""
+        if self._cmds:
+            line: str = self._cmds.pop(0)
+            return line.rstrip()
```

Comment:
`DTShInputFile` returns `line.rstrip()` because the lines we get from the file end with `\n`, I'm not sure it's useful to `rstrip()` the strings that come from `-c CMD` (possible leading and trailing spaces and tabs will be stripped from command lines by the parser):  

```suggestion
            return line
```